hicexplorer/hicAggregateContacts.py

```diff
@@ -392,133 +466,67 @@ def plot_aggregated_contacts(chrom_matrix, chrom_contact_position, cluster_ids,
                 X, Y = np.meshgrid(range(-M_half, M_half + 1),
                                    range(-M_half, M_half + 1))
                 Z = chrom_avg[chrom][cluster_number].copy()
-
                 img = ax.plot_surface(X, Y, Z, rstride=1, cstride=1, linewidth=0, cmap=cmap,
                                       vmax=vmax, vmin=vmin, edgecolor='none')
-
                 ax.set_zticklabels([])
                 if vmax is not None and vmax is not None:
                     ax.set_zlim(vmin, vmax)
-
-            if args.outFilePrefixMatrix:
+        
+        if args.outFilePrefixMatrix:
+            if args.genome : 
                 # save aggregate matrix values
                 if num_clusters == 1:
-                    output_matrix_name = "{file}_{chrom}.tab".format(file=args.outFilePrefixMatrix, chrom=chrom)
+                    output_matrix_name = "{file}.tab".format(file=args.outFilePrefixMatrix, chrom=chrom)
                 else:
-                    output_matrix_name = "{file}_{chrom}_cluster_{id}.tab".format(file=args.outFilePrefixMatrix,
-                                                                                  chrom=chrom, id=cluster_number + 1)
+                    output_matrix_name = "{file}_cluster_{id}.tab".format(file=args.outFilePrefixMatrix, chrom=chrom, id=cluster_number + 1)
                 np.savetxt(output_matrix_name, chrom_avg[chrom][cluster_number], '%0.5f', delimiter='\t')
-
+            else :
+                for idx, chrom in enumerate(chrom_matrix):
+                    for cluster_number, cluster_indices in enumerate(cluster_ids[chrom]):
+                        # save aggregate matrix values
+                        if num_clusters == 1:
+                            output_matrix_name = "{file}_{chrom}.tab".format(file=args.outFilePrefixMatrix, chrom=chrom)
+                        else:
+                            output_matrix_name = "{file}_{chrom}_cluster_{id}.tab".format(file=args.outFilePrefixMatrix, chrom=chrom, id=cluster_number + 1)
+                        np.savetxt(output_matrix_name, chrom_avg[chrom][cluster_number], '%0.5f', delimiter='\t')
+        
         cbar_x = plt.subplot(gs[-1, idx])
         fig.colorbar(img, cax=cbar_x, orientation='horizontal')
-
+    
     if args.disable_bbox_tight:
-        plt.savefig(args.outFileName.name, dpi=args.dpi)
+        plt.savefig(args.outFileName.name, dpi=100)
     else:
-        plt.savefig(args.outFileName.name, dpi=args.dpi, bbox_inches='tight')
-
+        plt.savefig(args.outFileName.name, dpi=100, bbox_inches='tight')
+    
     plt.close()
 
-
-def plot_diagnostic_heatmaps(chrom_diagonals, cluster_ids, M_half, args):
```

Comment:
Use more or less the deleted plot_diagnostic_heatmaps function, and try to integrate your changes to it.